inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/repositories/CampaignRepository.kt

```diff
@@ -0,0 +1,208 @@
+package com.rakuten.tech.mobile.inappmessaging.runtime.data.repositories
+
+import androidx.annotation.VisibleForTesting
+import com.google.gson.Gson
+import com.rakuten.tech.mobile.inappmessaging.runtime.InAppMessaging
+import com.rakuten.tech.mobile.inappmessaging.runtime.data.models.messages.Message
+import com.rakuten.tech.mobile.inappmessaging.runtime.data.responses.ping.CampaignData
+import com.rakuten.tech.mobile.sdkutils.PreferencesUtil
+import com.rakuten.tech.mobile.sdkutils.logger.Logger
+import org.json.JSONObject
+import java.lang.ClassCastException
+import java.lang.Integer.max
+
+internal interface CampaignRepositoryType {
+    val messages: List<Message>
+
+    val lastSyncMillis: Long?
+
+    /**
+     * Syncs [messages] with server.
+     */
+    fun syncWith(messageList: List<Message>, timestampMillis: Long)
+
+    /**
+     * Updates the [Message.isOptedOut] as true for the provided campaign.
+     */
+    fun optOutCampaign(campaign: Message): Message?
+
+    /**
+     * Decrements the number of [Message.impressionsLeft] for provided campaign Id in the repository.
+     */
+    fun decrementImpressions(id: String): Message?
+
+    /**
+     * Increments the number of [Message.impressionsLeft] for provided campaign Id in the repository.
+     */
+    fun incrementImpressions(id: String): Message?
```

Comment:
nit: add note that this is used for testing.

Reply:
Updated